Python/Product/PythonTools/PythonTools/Debugger/CustomDebuggerEventHandler.cs

```diff
@@ -83,5 +84,28 @@ public class CustomDebuggerEventHandler : IVsCustomDebuggerEventHandler110 {
                 "interpreter, please disable PGO.";
             MessageBox.Show(content, "PGO Is Not Supported", MessageBoxButton.OK, MessageBoxImage.Error);
         }
+
+        private void SetDebugOptions(IDebugEngine2 engine) {
+            var pyService = _serviceProvider.GetPythonToolsService();
+
+            var options = new StringBuilder();
+            if (pyService.DebuggerOptions.WaitOnAbnormalExit) {
+                options.Append(";" + AD7Engine.WaitOnAbnormalExitSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.WaitOnNormalExit) {
+                options.Append(";" + AD7Engine.WaitOnNormalExitSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.TeeStandardOutput) {
+                options.Append(";" + AD7Engine.RedirectOutputSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.BreakOnSystemExitZero) {
+                options.Append(";" + AD7Engine.BreakSystemExitZero + "=True");
+            }
+            if (pyService.DebuggerOptions.DebugStdLib) {
+                options.Append(";" + AD7Engine.DebugStdLib + "=True");
+            }
+
+            engine.SetMetric(AD7Engine.DebugOptionsMetric, options.ToString());

```

Comment:
Ah, one more reason to keep it intact. The current protocol allows for the flexibility of either overwriting an existing flag, or leaving it as is, and we actually use that to merge the values that flow via LaunchSuspended and via SetMetric.